src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/TextCompletionItemProvider.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion
+{
+    [Shared]
```

Comment:
If I understand the comments @ajaybhargavb has left me these aren't necessary when you're doing `services.AddSingleton`. I think there are some existing providers which incorrectly use these which is where you and I got the idea that they're needed.

Reply:
Removed, thanks for clarifying.